api/ondemand_vhds_rds_protocol.rst

```diff
@@ -0,0 +1,187 @@
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+During spontaneous updates configuration server will only send updates for
+virtual hosts that Envoy is aware of -- the configuration server needs to
+keep track of virtual hosts known to Envoy.
+
+Updates of non-virtual host fields
+==================================
+It might be required to update non-virtual host fields of route configuration
+while keeping virtual hosts as is. This would probably happen during
+spontaneous updates only. In such a case a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+containing a route configuration with an empty virtual hosts list can be used.
+
+Compatibility with Scoped RDS
+=============================
+Both approaches appear to be compatible with scoped RDS proposal:
+route_configuration name can still be used for vhost matching, but with scoped
+RDS configured it would point to a scoped route configuration.
+
+The xDS management server will also support the ability of Envoy to tell it
+when a resource hasn't been used and is safe to stop monitoring. The resources
+that can be removed include the base resources that xDS management server
+initially sent Envoy.
+
+Any future updates to this resource will be detected by xDS management server
+as part of the normal xDS flow and sent along with normal cluster updates.
+
+Full flow example
+==================
+
+.. figure:: diagrams/delta_rds_full_flow.svg
+   :alt: Delta RDS full flow
+
+Example xDS management server-generated RDS
+===========================================
+.. code:: yaml
+
+  '2001':
+    name: '2001'
+    virtual_hosts:
+    - name: s1http.none:2001
+      domains:
+      - s1http.none
+      - s1http.none:2001
+      routes:
+      - match:
+          PathSpecifier:
+            Prefix: "/"
+        Action:
+          Route:
+            ClusterSpecifier:
+              Cluster: outbound|2001||s1http.none
+            HostRewriteSpecifier:
+            timeout: 0
+            retry_policy:
+              retry_on: connect-failure,refused-stream,unavailable,cancelled,resource-exhausted
+              num_retries:
+                value: 10
+              retry_host_predicate:
+              - name: envoy.retry_host_predicates.previous_hosts
+                ConfigType:
+              host_selection_retry_max_attempts: 3
+              retriable_status_codes:
+              - 503
+            max_grpc_timeout: 0
+        decorator:
+          operation: s1http.none:2001/*
+        per_filter_config:
+          mixer:
+            fields:
+              disable_check_calls:
+                Kind:
+                  BoolValue: true
+              mixer_attributes:
```

Comment:
Why are there Mixer attributes in this example? Are they material? Generally prefer to avoid having Istio specifics creep into Envoy general purpose docs.

Reply:
Removed the example since it didn't add much value.